sdk/eventgrid/azure-eventgrid/azure/eventgrid/_models.py

```diff
@@ -0,0 +1,257 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+# pylint:disable=protected-access
+from typing import Optional
+from msrest.serialization import UTC
+import datetime as dt
+import uuid
+import json
+
+from ._generated.models._models import StorageBlobCreatedEventData, \
+    EventGridEvent as InternalEventGridEvent, \
+    CloudEvent as InternalCloudEvent
+
+class CloudEvent(InternalCloudEvent):   #pylint:disable=too-many-instance-attributes
```

Comment:
how is this different from the one in generated code? if it's the same there's no need to duplicate it.

Reply:
I have heard that convention is to add this in the models outside of generated code, even if it is pretty similar. I think the idea is that these are the models with exposed API, which are well separated from the generated. In this case, I also made a few changes, where I pass in default values for uuid and time to the generated CloudEvent model.